src/datalad_installer.py

```diff
@@ -849,6 +849,9 @@ def provide(
                 "install",
                 "pip @ git+https://github.com/pypa/pip",
             )
+        else:
+            # Ensure we have a recent pip
+            runcmd(installer.python, "-m", "pip", "install", "--upgrade", "pip")
```

Comment:
FTR: paranoid me had a concern if there would be no problem in case venv was created with --system-site-packages, in that upgrade here would fail to work.  But it seems to be all good
```
$> python -m pip install --upgrade pip
Requirement already satisfied: pip in ./venvtest/lib/python3.9/site-packages (20.3.4)
WARNING: Keyring is skipped due to an exception: Failed to unlock the keyring!
Collecting pip
  Using cached pip-21.3.1-py3-none-any.whl (1.7 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.3.4
    Uninstalling pip-20.3.4:
      Successfully uninstalled pip-20.3.4
Successfully installed pip-21.3.1
```